supertokens/supertokens.go

```diff
@@ -107,6 +107,10 @@ func supertokensInit(config TypeInput) error {
 }
 
 func defaultOnGeneralError(err error, req *http.Request, res http.ResponseWriter) {
+	w := MakeDoneWriter(res)
+	if w.IsDone() {
+		return
+	}
```

Comment:
Now that i think of this, should we do this check before we call the onGeneralError handler? Cause the user can override this and they will not do this check before sending a 500 to the client.. so then in that case, it will cause issues. 

What do you think?